public/doc/openapi-3-v1.1.json

```diff
@@ -3038,7 +3038,136 @@
             "type": "object",
             "title": "Metadata",
             "description": "JSON Metadata about the portfolio item",
+            "readOnly": true,
+            "properties": {
+              "user_capabilities": {
+                "type": "object",
+                "description": "A list of permissions that the current user has access to on this portfolio item"
+              },
+              "orderable": {
```

Comment:
Was there thought to making `orderable` part of `user_capabilities`?  Is there a reason it needs to be outside of that object?

Reply:
@gmcculloug I thought we determined that `user_capabilities` were more like "permissions" or "do you have the authority to do this action?" versus "will this work if I do it right now?"

Reply:
I am not sure if we should be creating custom response object in the openapi.json we can use a field in the metadata, user_capabilites is only a one part of the metadata, we can add extra metadata fields. As far as versioning is concerned we have a partial solution, we started with controllers, then we added services and now we are into models but models don't support versioning yet. When we add a new field/metadata should the models be versioned.

Reply:
> I am not sure if we should be creating custom response object in the openapi.json we can use a field in the metadata, user_capabilites is only a one part of the metadata, we can add extra metadata fields

@mkanoor I'm not sure what you mean here, isn't what I added just an extra metadata field? 

I would prefer if the openapi.json was able to govern the `as_json` serialization, but I don't think that's feasible for nested hashes. Common currently does not attempt to serialize any fields that aren't in the openapi.json, which is great if you're trying to save an expensive method call at the top attribute level, but you can't really do that with nested hash keys. We could throw out the data if it's not in openapi.json, but it's still gonna calculate it so that needs to be handled by the model/controller/service itself.

I don't think I want to version models at this time. You then have to potentially worry about either creating a new database, and somehow keeping both versions supported, or you have the same database but need to make sure all the changes you make are forward and backward compatible. Not to mention the deployment intricacies and/or if we now need multiple pods running different versions, etc. I think "working around" wanting versioning in the model is going to be cheaper (in the short term and long term) than attempting to version them.

Maybe the solution is that we just add "orderable" as an attribute. The only problem then is that our versioned service would need to be called directly from the model. Instead, I would propose that the `PortfolioItemOrderable` service be moved to lib, and the simple fact that "orderable" exists as something that is returned is what is versioned, not the logic. `PortfolioItemOrderable` already calls another lib service (`SurveyCompare`), and it's pure logic based on figuring out whether or not a portfolio item is orderable. In theory all of this logic *could* be done directly on the portfolio item itself, it's just cleaner to have in a separate class, which sounds like a good candidate for a lib service to me. It kinda sucks that we're not really utilizing the "metadata" attribute, since this additional information feels exactly like what we built it for, but versioning and/or separating "metadata" to be returned differently for different API calls doesn't seem very simple.

Reply:
@eclarizio the metadata is a free format object with no properties. It allows us to put all kinds of data in it and is not limited to specific attributes. The way your PR is implementing it is limiting it to user_capabilities and orderable. They shouldn't be getting listed here. None of the properties/attributes in the metadata is queryable and it's purely for informational purposes.

Reply:
@mkanoor For this PR, it's not limiting it at all. You can shove as much data as you want in the metadata attribute still. The openapi.json change is simply documentation that shows an example of what a user can expect to receive if they make this call. If they get back more than this, that's kind of our fault for not providing it as expected in the openapi json.